### PR TITLE
Add Jetpack powered banner to Reader Search

### PR DESCRIPTION
**Pull request by Gio2018**
Fixes #NA

This PR adds a "Jetpack powered" banner in the reader search of the WordPress app (see screenshots).
The banner won't be visible on iPhone landscape.
This PR also removes (temporarily) the "Search WordPress for a site or a post" label that appeared below the "Clear search history" button.

Newer iPhone portrait | iPhone landscape | older iPhone portrait with larger fonts
-- | -- | --
 |  | 

To test:
- build/run the `WordPress` target from this branch
- Go to Reder and select search in the top-right corner
- on iPad and iPhone portrait: make sure that the "Jetpack powered" banner appears
- Rotating to landscape on iPhone (or starting the app in landscape) make sure that the banner does not appear
- Stop the execution and build/run the `Jetpack` target from this branch
- Repeat the above steps and make sure that the "Jetpack powered" banner does not show in any step.

## Regression Notes
1. Potential unintended areas of impact
Low, it adds an `inputAccessoryView` to the `serchBar` in `ReaderSearchViewController`
This PR also adjusts heights calculation in `ReaderSearchSuggestionsViewController` and removes a label in `ReaderSearchViewController`

2. What I did to test those areas of impact (or what existing automated tests I relied on)
test on several devices and font sizes

3. What automated tests I added (or what prevented me from doing so)
none
PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] ~~I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.~~ Not released.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Reader/ReaderSearchViewController.swift [hunk 1]
```diff
@@ -169,13 +173,35 @@ import Gridicons
...
+    /// hides the Jetpack powered banner on iPhone landscape
+    private func hideBannerViewIfNeeded() {
+        guard AppConfiguration.isWordPress, FeatureFlag.jetpackPowered.enabled else {
+            return
+        }
+        let height = UIApplication.shared.mainWindow?.frame.size.height ?? 0
+        // maximum height of any iPhone in landscape. iPads and portrait orientations
+        // all have heights greater than this one.
+        let maximumLandscapeHeight: CGFloat = 428
+        bannerView.isHidden = height <= maximumLandscapeHeight
```
Comment (sla8c) [inline WordPress/Classes/ViewRelated/Reader/ReaderSearchViewController.swift]: Perhaps a cleaner solution would be to use size classes here?

For example something like this:
```
if traitCollection.verticalSizeClass == .compact {
    bannerView.isHidden = true
} else {
    bannerView.isHidden = false
}
```

**Comment by twstokes** [comment]
👋 @Gio2018, we may need to do some tweaks since [this PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/19062) added the banner to some areas of Reader (including the main search view).

Running the latest on this branch I see:

File: WordPress/Classes/ViewRelated/Reader/ReaderSearchViewController.swift (hunk 1 above)
Comment (Gio2018) [reply WordPress/Classes/ViewRelated/Reader/ReaderSearchViewController.swift]: Indeed we should use the size class 🤦 , thanks for bringing this up @sla8c !

**Comment by Gio2018** [comment]
> 👋 @Gio2018, we may need to do some tweaks since [this PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/19062) added the banner to some areas of Reader (including the main search view).
> 

Thanks for bringing this up @twstokes ! I am now hiding the bar from reader stream (if present). That issue though is still there on newer iPhones. It's the accessory view still showing up when an hardware keyboard is connected. Does not happen on iPad. Not sure if we want to add logic to handle that, or treat it as an edge case, considering that it does not seem to happen on iPad? Wdyt?

**Review by sla8c — APPROVED** [review]

**Comment by sla8c** [comment]
@Gio2018 FYI I've noticed unit tests were failing; I re-triggered in case it was a build kite issue

**Comment by twstokes** [comment]
👋 @Gio2018 - how are we treating Reader search for self-hosted sites not logged into WordPress.com?

Since being logged into .com + Jetpack would enable the ability to star / favorite / re-blog, should we not show the banner in this scenario?

**Comment by twstokes** [comment]
> > 👋 @Gio2018, we may need to do some tweaks since [this PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/19062) added the banner to some areas of Reader (including the main search view).
> 
> Thanks for bringing this up @twstokes ! I am now hiding the bar from reader stream (if present). That issue though is still there on newer iPhones. It's the accessory view still showing up when an hardware keyboard is connected. Does not happen on iPad. Not sure if we want to add logic to handle that, or treat it as an edge case, considering that it does not seem to happen on iPad? Wdyt?

Thanks for the info! It sounds like we're probably OK to keep it like it is, then. 👍

**Review by twstokes — APPROVED** [review]
LGTM @Gio2018 🚀  - I just had [one comment](https://github.com/wordpress-mobile/WordPress-iOS/pull/19059#issuecomment-1190632330) on if this should be shown for self-hosted users who aren't logged into .com / connected to Jetpack.

**I tested:**
- Jetpack app
- iPad
- Dark mode
- a11y
- Different orientations

**Comment by Gio2018** [comment]
> 👋 @Gio2018 - how are we treating Reader search for self-hosted sites not logged into WordPress.com?
> 
> Since being logged into .com + Jetpack would enable the ability to star / favorite / re-blog, should we not show the banner in this scenario?

Great point @twstokes ! As discussed on Slack, this will be covered once we move to the `single flag` that performs all the due checks.

**Comment by Gio2018** [comment]
> @Gio2018 FYI I've noticed unit tests were failing; I re-triggered in case it was a build kite issue

Thanks for re-running the tests @sla8c ! it seems that they succeeded this time.
